### ISSUE TITLE
ImportError with 'Self' from 'typing' in Python 3.10

### ISSUE THREAD

**Issue by runker54** [alecsandrei/PySAGA-cmd#1]
# Issue Description
I encountered an ImportError when trying to import the SAGA class from the PySAGA_cmd library. The error message indicates that it cannot import Self from the typing module.

# Environment
Operating System: WIN11
Python Version: 3.10
PySAGA_cmd Version: 1.0.6
# Steps to Reproduce
1. Install the PySAGA_cmd library.

2. Attempt to execute the following code:
```python
from PySAGA_cmd import SAGA
saga = SAGA(r"D:\Software\saga-9.3.0_x64\saga_cmd.exe")
```
```
ImportError: cannot import name 'Self' from 'typing' (D:\Programs\Python\Python310\lib\typing.py)

```

**Thank you for your time and effort, looking forward to your response.**

**Comment by alecsandrei** [comment]
Hello,

Thank you for this issue. Will implement a fix later today. It's because Self is only available from Python 3.11 onwards.

**Comment by alecsandrei** [comment]
the commit bac37ea solved the issue
